index.html

```diff
@@ -5,7 +5,7 @@
     <meta name="viewport" content="width=device-width, initial-scale=1.0" />
     <title>US States Quiz</title>
     <link rel="stylesheet" href="styles.css" />
-    <script src="./js/index.js" defer></script>
+    <script src="./js/index.js" type="module"></script>
```

Comment:
This could be the mistake. You tell the HTML, that your index.js is in the folder js, but you have no js folder.

Your index.js is on the same level as your index HTML.